Sources/StreamChat/Utils/Database/DatabaseItemConverter.swift

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+/// Converts database models to immutable value types by reusing existing items.
+enum DatabaseItemConverter {
```

Comment:
Should we add some test coverage to this component To make sure the logic is correct?

Reply:
I added tests one layer above since I needed DTOs and models. Therefore, this is covered indirectly.

Reply:
I see; I was thinking if this was possible to test as Unit Test only, but thanks for clarifying 👍